File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/achievement/AchievementFinishedInformationMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AchievementFinishedInformationMessage extends AchievementFinishedMessage {
  override val messageId = 6381

  protected trait Message extends AchievementFinishedMessage.Message with Serializable {
    override val messageId = AchievementFinishedInformationMessage.messageId

    val name: String
    val playerId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeString(name)
      buf.writeInt(playerId)
    }

  }

  protected def create(id: Short, finishedlevel: Short, name: String, playerId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val id = buf.readShort()
    val finishedlevel = buf.readShort()
    val name = buf.readString()
    val playerId = buf.readInt()

    create(id, finishedlevel, name, playerId)
  }

}

object AchievementFinishedInformationMessage extends AchievementFinishedInformationMessage {
  case class MessageImpl(id: Short, finishedlevel: Short, name: String, playerId: Int) extends Message
  type Target = MessageImpl

  override def create(id: Short, finishedlevel: Short, name: String, playerId: Int) = MessageImpl(id, finishedlevel, name, playerId)

  def apply(id: Short, finishedlevel: Short, name: String, playerId: Int) = create(id, finishedlevel, name, playerId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AchievementFinishedInformationMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AchievementFinishedInformationMessage extends AchievementFinishedMessage {
    public static final int MESSAGE_ID = 6381;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public String name;
    public int playerId;
    
    public AchievementFinishedInformationMessage() { }
    
    public AchievementFinishedInformationMessage(short id, short finishedlevel, String name, int playerId) {
        super(id, finishedlevel);
        this.name = name;
        this.playerId = playerId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeString(name);
        buf.writeInt(playerId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        name = buf.readString();
        playerId = buf.readInt();
        if (playerId < 0)
            throw new RuntimeException("Forbidden value on playerId = " + playerId + ", it doesn't respect the following condition : playerId < 0");
    }
    
}
